src/bucket/BucketController.ts

```diff
@@ -153,6 +181,60 @@ export class BucketController {
       this.allowDownload = true;
     }
 
+    // default sts usage to false
+    if (this.stsApiUrl === undefined) {
+      this.stsEnabled = false;
+    } else {
+      this.stsEnabled = true;
+    }
+
+    // collect sts credentials if needed
+    if (this.stsEnabled) {
+      var httpHeaders;
+      if (this.stsHeaderName === undefined || this.stsHeaderValue === undefined) {
+        httpHeaders = {
+          headers: {
+            'Content-Type': 'application/json'
+          }
+        };
+      } else {
+        var name = this.stsHeaderName;
+        httpHeaders = {
+          headers: {
+            'Content-Type': 'application/json',
+            name: this.stsHeaderValue
+          }
+        };
+      }
+
+      var result: ng.IPromise<any> = this.httpService.post(
+        this.stsApiUrl,
+        {},
+        httpHeaders)
+      .then((response: any): ng.IPromise<any> => this.refreshSTSCredentials(response));
+    } else {
+      // create the backend
+      if (this.backendName === 's3') {
+        this.backend = new AmazonS3Backend(
```

Comment:
This logic is unnecessarily convoluted. You can very easily build this as a chain of promises so you're not creating the backend in two places. Also it's confusing to have initialization logic outside of `$onInit`.

Reply:
The reason I have to create the backend outside of `$onInit` is because in a future card I will be implementing auto-refreshing sts credentials.  In that situation, the backend will have to be created outside of `$onInit` irregardless.
Since the http post runs asynchronously and not 100% of the time, I had to have the backend created in multiple locations.

Moving this into a top-level component might fix the issue, but I have not had the time to debug that yet.  (and we would still have to regenerate the backend when the credentials refresh)